Mysql cookbook v8

gsreynolds opened the pull request:
Hi! Saw #5 and that the cookbook was blocked on Chef 13 compat due to mysql2_chef_gem cookbook v1 and just wanted to contribute a few fixes you might find useful. Hope this helps.

The ohai plugin was broken on debian reporting:

```
  "mysql": {
    "installed_version": null
  }
```

There also seems to be an issue with ohai complaining about /etc/chef/ohai/plugins not existing, thus breaking the `its(:stderr) { should be_empty }` examples, I've added a workaround.

gsreynolds committed:
Update mysql cookbook to version 8

gsreynolds committed:
Update ohai cookbook to version 5

gsreynolds committed:
Fix ohai plugin on debian & ubuntu, mysqld_bin is /usr/sbin/mysqld

gsreynolds committed:
Silence extraneous stderr output from ohai
"INFO: The plugin path /etc/chef/ohai/plugins does not exist. Skipping..."

gsreynolds committed:
Travis is executing TESTS=“style unit” using debian family docker image, thus shell out becomes /usr/sbin/mysqld

zuazo commented:
Thanks @gsreynolds! Nice work. Really appreciated.

Give me a couple of days to review this properly.

zuazo commented on test/cookbooks/mysql_tuning_test/recipes/ohai_plugin.rb:
```diff
@@ -19,5 +19,11 @@
+# "INFO: The plugin path /etc/chef/ohai/plugins does not exist. Skipping..."
+directory '/etc/chef/ohai/plugins' do
+  recursive true
+end
```
Maybe we should move this to the `mysql_tuning::ohai_plugin` recipe, shouldn't we?

gsreynolds replied:
I'm not sure that we should. It's very much a workaround for tests only. Under Test Kitchen, Ohai's plugin directory is /tmp/kitchen/ohai/plugins/, but Ohai still erroneously emits stderr `"INFO: The plugin path /etc/chef/ohai/plugins does not exist. Skipping..."` because it assumes that path (https://github.com/chef/ohai/blob/7254fff3cbcc4ea489259c7d7a86e9c317148bdf/lib/ohai/config.rb#L37)

Even setting the following in .kitchen.yml and passing the `ohai` command the client.rb config doesn't fix it.
```
provisioner:
  name: chef_zero
  client_rb:
    ohai.directory: /tmp/kitchen/ohai/plugins
```

If anything, I think it would need to be addressed upstream in https://github.com/chef/ohai/.

zuazo reviewed (changes requested):
OK, nice work :wink:

Would it be possible to fix the unit tests? I mean, if they are not trivial to fix, at least we should put them in pending status (`xit`).

gsreynolds committed:
Load Ohai gem from the Chef omnibus install and use directly, rather than via ohai command line.

Better fix for extraneous stderr from ohai "INFO: The plugin path /etc/chef/ohai/plugins does not exist. Skipping..."

gsreynolds committed:
Minor fixes to unit tests, xit pending status for mysql_tuning_cnf resource specs

gsreynolds commented:
Thanks! 

I've pushed a further commit that uses the Ohai gem directly rather than the `ohai` command line, which more neatly works around the stderr issue. I've also fixed some of the unit tests and `xit` others, so it should hopefully pass Travis now. It seems that for some reason ChefSpec isn't stepping into the mysql_tuning and mysql_tuning_cnf resources, so the template etc resources expected in `test/unit/resources/cnf_spec.rb` aren't in the resource collection. I don't know why, I'm not entirely sure this is because of the PR and I don't have the bandwidth to look into it further, sorry!

~~I ran `cookstyle -a` against the repo, but I see from Travis that the CI setup uses plain Rubocop so now the linting disagrees with it. I don't want to change any of your CI setup, so I leave that up to you :wink:. Let me know if you want me to drop that commit and force push without it.~~
I have dropped this commit and replaced with one with Rubocop fixes instead so we can see Travis green!

gsreynolds committed:
Rubocop fixes

zuazo commented:
@gsreynolds thanks for your work!

I merged it manually after reverting some of the changes you made to the tests in 1fe27f43640abaed6bddbcd37d1fc40cb1166b4c. In order to fix the unit tests, I just had to force the installation of Chef 12 in the Travis configuration file. That avoids running the tests using Chef 13, which breaks some ChefSpec tests when working with LWRPs.

zuazo commented:
Released in [`0.8.0`](https://github.com/zuazo/mysql_tuning-cookbook/releases/tag/0.8.0).

gsreynolds commented:
@zuazo thanks for the contributor mention! Makes sense to revert those ChefSpec changes and run them under Chef 12 for now.
